c3pzero_ignition/src/odom2tf_broadcaster_node.cpp

```diff
@@ -0,0 +1,57 @@
+/*******************************************************
+ * Copyright (C) 2022 Hyundai America Technical Center, Inc. (HATCI)
+ *
+ * This file can not be copied and/or distributed without the express
+ * permission of Hyundai America Technical Center, Inc. (HATCI)
+ *******************************************************/
+
+#include <memory>
+
+#include "geometry_msgs/msg/transform_stamped.hpp"
+#include "nav_msgs/msg/odometry.hpp"
+#include "tf2_ros/transform_broadcaster.h"
+#include <rclcpp/rclcpp.hpp>
+
+using std::placeholders::_1;
+
+/**
+ * @brief Simple class that broadcasts tf from the odometry message
+ *
+ */
+class TfBroadcaster : public rclcpp::Node
+{
+public:
+  TfBroadcaster() : Node("tf_broadcaster")
+  {
+    subscription_ = this->create_subscription<nav_msgs::msg::Odometry>(
+        "odom", 10, std::bind(&TfBroadcaster::topic_callback, this, _1));
+  }
+
+private:
+  void topic_callback(const nav_msgs::msg::Odometry::SharedPtr msg)
+  {
+    tf_broadcaster_ = std::make_shared<tf2_ros::TransformBroadcaster>(shared_from_this());
+    geometry_msgs::msg::TransformStamped transformStamped;
```

Comment:
How about we make this a private member so we only create it once and set the fields whenever we process a callback?

Reply:
Done